dirsrvtests/tests/suites/subentries/subentries_test.py

```diff
@@ -58,156 +54,108 @@ def encodeControlValue(self):
-            entries = topo.standalone.search_ext_s("ou=people," +
-                DEFAULT_SUFFIX,
-                ldap.SCOPE_SUBTREE,
-                search_filter,
-                serverctrls=[request_ctrl])
-        else:
-            entries = topo.standalone.search_s("ou=people," + DEFAULT_SUFFIX,
-                ldap.SCOPE_SUBTREE,
-                search_filter)
-        if searchEntries and searchSubentries:
-            pass
-        else:
-            if (useControl and not controlValue) or \
-                (not useControl and searchEntries):
-                for entry in entries:
-                    assert ensure_bytes("ldapsubentry") \
-                        not in entry.getValues('objectclass')
-            elif (useControl and controlValue) or \
-                (not useControl and searchSubentries):
-                for entry in entries:
-                    assert ensure_bytes("ldapsubentry") \
-                        in entry.getValues('objectclass')
-        return len(entries)
-    except ldap.LDAPError as e:
-        log.fatal('Search failed: ' + e.args[0]['desc'])
-        assert False
-
-@pytest.mark.skipif(not has_subentries_control(), reason="This test is only\
-         required when LDAP Subentries control is supported.")
-def test_subentries_control(topo):
-    """Test LDAP Subentries (RFC 3672) control
-
-    :id: 62e00650-979d-456f-bc53-e4b6af32fd0e
+    return "1.3.6.1.4.1.4203.1.10.1" in rdse.get_supported_ctrls()
+
+
+@pytest.fixture(scope="module")
+def setup_test_entries(topo, request):
+    """
+    Add nentries entries and nentries subentries.
+    """
+    users = []
+    # Add normal entries.
+    user = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    for i in range(0, nentries):
+        user1 = user.create_test_user(uid=i)
+        users.append(user1)
+
+    # Add subentries.
+    for i in range(nentries, nentries * 2):
+        user1 = user.create_test_user(uid=i)
+        user1.add("objectclass", "ldapsubentry")
+        users.append(user1)
+
+    def fin():
+        for user in users:
+            user.delete()
+
+
+    request.addfinalizer(fin)
+
+
+nentries = 5
+search_entries = '(objectclass=inetorgperson)'
+search_subentries = '(objectclass=ldapsubentry)'
+
+# Test matrix parameters
+searches = [
+            # Search with subentries control visibility TRUE
+            (search_entries, True, True, nentries, True),
+            # Search with subentries control visibility FALSE
+            (search_entries, True, False, nentries, False),
+            # Search for normal entries
+            (search_entries, False, None, nentries, False),
+            # Search for subentries
+            (search_subentries, False, None, nentries, True),
+            # Search for normal entries and subentries
+            (f'(|{search_entries}{search_subentries})', False, None,
+                nentries * 2, None)
+        ]
+
+@pytest.mark.parametrize('search_filter, use_control, controlValue,'\
```

Comment:
i see what you really meant by parametrizing now. thats much neater altho it would add/delete the same entries for each invokation scenario.

Reply:
The scope of the `setup_test_entries` fixture is "module", so the entries are created at the beginning of the first test that uses it in the module and removed only after the last one. In case of the parametrized test case, fixture is not executed on each new parameter. You can check the access logs to see what I mean.

Reply:
ah ok, thats good then and i'm sorry i'm just not familiar with pytest at all and this is literally my 1st time looking into it.